saleor/graphql/account/mutations/authentication/set_password.py

```diff
@@ -79,6 +80,11 @@ def _set_password_for_user(cls, email, password, token):
             password_validation.validate_password(password, user)
         except ValidationError as error:
             raise ValidationError({"password": error})
+        fields_to_save = ["password", "updated_at"]
         user.set_password(password)
-        user.save(update_fields=["password", "updated_at"])
+        if not user.is_confirmed:
```

Comment:
Maybe it's worth leaving here a small comment that will explain that to reset the password, a user needs to process the token which was sent separately, so now we are sure, that he has access to email account 🤔

Reply:
Done